File: src/network/packet/impl/ServerChangeStatusPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\CloudboundPacket;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\util\PacketData;
use pocketcloud\cloud\server\CloudServerManager;
use pocketcloud\cloud\server\util\ServerStatus;

final class ServerChangeStatusPacket extends CloudPacket implements CloudboundPacket {

    public function __construct(
        private string $serverUuid = "",
        private ?ServerStatus $status = null
    ) {}

    public function handle(ServerClient $client): void {
        if ($client->hasServer()) {
            CloudServerManager::getInstance()->get($this->serverUuid)?->setServerStatus($this->status);
        }
    }

    public function encodePayload(PacketData $packetData): void {}

    public function decodePayload(PacketData $packetData): void {
        $packetData->readAllTypeSafe([&$this->serverUuid, &$this->status], [fn() => $packetData->readString(), fn() => $packetData->readServerStatus()]);
    }

    public function getServerUuid(): string {
        return $this->serverUuid;
    }

    public function getStatus(): ?ServerStatus {
        return $this->status;
    }

    public static function create(string $serverUuid, ServerStatus $status): self {
        return new self($serverUuid, $status);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/ServerChangeStatusPacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.network.packet.AuthenticatedPacket;
import de.pocketcloud.network.packet.CloudboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.network.packet.data.PacketData;
import de.pocketcloud.cloud.server.CloudServerManager;
import de.pocketcloud.cloud.server.util.ServerStatus;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class ServerChangeStatusPacket extends CloudPacket implements CloudboundPacket, AuthenticatedPacket {

    private String serverUuid;
    private ServerStatus status;

    public ServerChangeStatusPacket(String serverUuid, ServerStatus status) {
        this.serverUuid = serverUuid != null ? serverUuid : "";
        this.status = status;
    }

    @Override
    public void handle(@NotNull ServerClient client) {
        CloudServerManager.instance().get(serverUuid).ifPresent(server -> server.setStatus(status));
    }

    @Override
    public void encodePayload(PacketData packetData) {}

    @Override
    public void decodePayload(PacketData packetData) {
        this.serverUuid = packetData.readString();
        this.status = packetData.readEnum(ServerStatus.class);
    }

    public static ServerChangeStatusPacket create(String serverUuid, ServerStatus status) {
        return new ServerChangeStatusPacket(serverUuid, status);
    }
}
